DocumentDataAPI/DocumentDataAPI/Controllers/DocumentController.cs

```diff
@@ -0,0 +1,208 @@
+        [HttpPut]
+        [Route("PutDocument")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<DocumentModel> PutDocument([FromBody] DocumentModel document)
+        {
+            try
+            {
+                return _repository.Add(document) == 0
+                    ? Problem("No rows were added")
+                    : Ok(_repository.Get(document.Id));
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Unable to add document.\n{e.Message}");
+                return Problem(e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves a list of all documents from the database.
+        /// </summary>
+        /// <response code="200">Success: A list of all documents</response>
+        /// <response code="404">Not Found: Nothing is returned.</response>
+        /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+        [HttpGet]
+        [Route("Get")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<IEnumerable<DocumentModel>> GetAll()
+        {
+            try
+            {
+                List<DocumentModel> result = _repository.GetAll().ToList();
+                return result.Count == 0 ?
+                    NotFound() :
+                    Ok(result);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Unable to fetch documents\n{e.Message}");
+                return Problem(e.Message);
+            }
+
+        }
+
+        /// <summary>
+        /// Retrieves a document based on the document id.
+        /// </summary>
+        /// <response code="200">Success: A document for the given id.</response>
+        /// <response code="404">Not Found: Nothing is returned.</response>
+        /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+        [HttpGet]
+        [Route("Get/{id:int?}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<DocumentModel> GetById(int id)
+        {
+            try
+            {
+                DocumentModel? result = _repository.Get(id);
+                return result == null ?
+                    NotFound() :
+                    Ok(result);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Unable to fetch document with id: {id}.\n{e.Message}");
```

Comment:
Samme som i andre kommentarer. Bemærk at man ikke bruger string interpolation ($) i loggeren, det gøres som vist i min suggestion
```suggestion
                _logger.LogError(e, "Unable to fetch document with id: {id}", id);
```